base/src/test/java/com/trackray/base/utils/RegexUtilTest.java

```diff
@@ -0,0 +1,83 @@
+package com.trackray.base.utils;
+
+import com.trackray.base.utils.RegexUtil;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class RegexUtilTest {
+
+  @Test
+  public void testExtractStr() {
+    Assert.assertNull(RegexUtil.extractStr("foobar", null));
+    Assert.assertNull(RegexUtil.extractStr(null, "^.+(bar)"));
+    Assert.assertEquals("bar", RegexUtil.extractStr("foobar", "^.+(bar)"));
+    Assert.assertEquals("bar", RegexUtil.extractStr("foobar", "^.+(bar)", false));
```

Comment:
What does false do differently?

Reply:
If true, it attempts to strip out any html escape characters first, before processing the remaining string. But that's in another module which i'm not testing here.